### PR TITLE
 #167288153 get user test questions 

### PR DESCRIPTION
**Pull request by Ochowo**
### What does this PR do?

* Get the test question for each module

### Description of Task to be completed?

* Create action
* Create Reducers
* Add component
* Write the test 
### Any background context you want to provide?

N/A

### What are the relevant Github issues?

N/A

### Questions:

N/A

### Screenshot:
Screenshot 2019-07-30 at 3 17 25 PM

### CODE REVIEW

File: src/components/Questions/Questions.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      <div id="psub">
+        {' '}
+        <Button className="psub">Submit</Button>
+      </div>
+    </div>
+  </section>
+);
+
+TestQuestions.propTypes = {
+  testQuestions: PropTypes.any,
```
Comment (fortunee) [inline src/components/Questions/Questions.js]: Please let's be more explicit on the type of `testQuestions` here

Comment (fortunee) [reply src/components/Questions/Questions.js]: And I think some of these props are required for this component, please specify the appropriate prop types

File: src/views/ModuleTest/ModuleTest.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+class TestQuestions extends React.Component {
+  componentDidMount() {
+    const { getModuleTest, match } = this.props;
+    const { moduleId } = match.params;
+    getModuleTest(moduleId);
+  }
+
+  handleChange(e) {
+    const { name } = e.target;
+    const { value } = e.target;
```
Comment (fortunee) [inline src/views/ModuleTest/ModuleTest.js]: Is there any reason we can't destructure both `name` and `value` this in a single line? 🤔 

```js
const { name, value } = e.target;
```

File: src/views/ModuleTest/ModuleTest.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    const { test, isLoading } = this.props;
+    return (
+      <Fragment>
+        <Navbar />
+        <section className="sec-question">
+          <div className="que-container">
+            <Questions
+              testQuestions={test}
+              loading={isLoading}
+              onChange={e => this.handleChange(e)}
```
Comment (fortunee) [inline src/views/ModuleTest/ModuleTest.js:33]: You know you can do this as well 👇 

```js
onChange={this.handleChange}
```

File: src/views/ModuleTest/ModuleTest.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        </section>
+        <Footer />
+      </Fragment>
+    );
+  }
+}
+
+TestQuestions.propTypes = {
+  getModuleTest: PropTypes.func.isRequired,
+  test: PropTypes.any,
```
Comment (fortunee) [inline src/views/ModuleTest/ModuleTest.js]: Define explicit types for props `test` here

File: src/views/ModuleTest/ModuleTest.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  moduleId: state.modules.singleModule.moduleId,
+});
+
+export const mapDispatchToProps = dispatch => bindActionCreators(
+  {
+    getModuleTest: getTest,
+  },
+  dispatch,
+);
+export { TestQuestions };
```
Comment (fortunee) [inline src/views/ModuleTest/ModuleTest.js:67]: Why the 2 different exports on `TestQuestions`?

**Review by fortunee — APPROVED** [review]

**Review by fortunee — CHANGES_REQUESTED** [review]
Good job @Ochowo Just a few comments to improve on

File: src/views/ModuleTest/ModuleTest.js (hunk 5 above)
Comment (Ochowo) [reply src/views/ModuleTest/ModuleTest.js:67]: @fortunee the default export is the redux connected version of the component. I need this second export to be able to test the rendering of the component.

Comment (fortunee) [reply src/views/ModuleTest/ModuleTest.js:67]: Doesn't make sense to me

**Review by fortunee — APPROVED** [review]
👍

**Pull request merged**